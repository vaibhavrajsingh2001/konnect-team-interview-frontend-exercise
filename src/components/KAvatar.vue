<template>
  <img
    :alt="name"
    class="k-avatar"
    :src="avatarUrl"
    :title="title"
  >
</template>
<script lang="ts">
import { defineComponent } from 'vue'

export default defineComponent({
  name: 'KAvatar',
  props: {
    avatarUrl: {
      type: String,
      required: true,
    },
    name: {
      type: String,
      required: true,
    },
    email: {
      type: String,
      required: true,
    },
  },
  computed: {
    title(): string {
      return `${this.name} <${this.email}>`
    },
  },
})
</script>
<style lang="scss">
.k-avatar {
  height: 3.6rem;
  width: 3.6rem;
  border-radius: 50%;
}
</style>
