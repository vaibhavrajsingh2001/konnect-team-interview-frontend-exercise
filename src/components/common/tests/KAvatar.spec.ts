import { renderToString } from '@vue/test-utils'
import { it, describe, expect } from 'vitest'
import KAvatar from '@/components/common/KAvatar.vue'

describe('KAvatar', () => {
  it('renders correctly', async () => {
    const wrapper = await renderToString(KAvatar, {
      props: {
        avatarUrl: 'https://example.com/image.jpg',
        name: 'Example image',
        email: 'sample@sample.com',
      },
    })
    expect(wrapper).toMatchSnapshot()
  })
})
