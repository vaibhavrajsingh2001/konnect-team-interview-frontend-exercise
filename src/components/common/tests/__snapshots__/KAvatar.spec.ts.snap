// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`KAvatar > renders correctly 1`] = `"<img alt="Example image" class="k-avatar" src="https://example.com/image.jpg" title="Example image &lt;sample@sample.com&gt;">"`;
